include/avl.h

```diff
@@ -21,32 +21,75 @@
 
 #include "common.h"
 
+
 #pragma once
 
 typedef struct avl {
+	int keyNode;
 	int data;
 	struct avl *left;
 	struct avl *right;
 	int height;
 } avl_t, *avl_p, **avl_pp;
 
+
+
 /* Generate AVL tree from an array of input values */
 avl_pp generate_avl(int *arr, int len);
 
 /* Initialize an AVL tree */
 avl_pp init_avl(void);
 
 /* Insert a node in AVL tree */
-bool insert_avl_node(avl_pp head, int val);
+bool insert_avl_node(avl_pp head, int key, int data);
+
 
 /* Delete a node from AVL tree */
-bool delete_avl_node(avl_pp head, int val);
+bool delete_avl_node(avl_pp head, int key);
 
 /* Destroy the tree */
 int destroy_avl(avl_pp head);
+// avl will be destroy isn't necessary sincronize it (is better not destroy)
 
 /* Print a tree in preorder */
 int print_avl(avl_p root, avl_p parent);
 
 /* Traverse tree in BFS to find a given value */
-bool search_BFS_avl(avl_pp root, int val, bool stop);
+int search_BFS_avl(avl_pp root, int key);
```

Comment:
Why was the `stop` param removed? I think we used to set it to FALSE to print the AVL in BFS traversal.

Reply:
I removed it because now search return value of node searched by  key, and continue for the next node, i thinked there wasn't inteligent.
But now i discover you think it to set it to FALSE to print the AVL in BFS traversal.
So now i restore it and also add show parameter to decide if show debug info or not.
what you think?

Reply:
Yes, we have to have a way to print the whole tree.